osioauth/tokens.go

```diff
@@ -0,0 +1,129 @@
+package osioauth
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"net/url"
+
+	"github.com/fabric8-services/fabric8-tenant/configuration"
+	"github.com/pkg/errors"
+)
+
+type authAccessToken struct {
+	AccessToken string `json:"access_token,omitempty"`
+}
+
+type authEerror struct {
+	Code   string `json:"code,omitempty"`
+	Detail string `json:"detail,omitempty"`
+	Status string `json:"status,omitempty"`
+	Title  string `json:"title,omitempty"`
+}
+
+type errorResponse struct {
+	Errors []authEerror `json:"errors,omitempty"`
+}
+
+func GetAuthAccessToken() (string, error) {
```

Comment:
Code to an interface.

have something like a 

```
type TokenService interface{
     GetAuthAccessToken() (string, error) 
     GetOpenShiftToken() (string,error)
}

type TokenClient struct{}

func (c *TokenClient)GetAuthAccessToken() (string, error) {
   // implement the method.
}
```

This will help you inject the service wherever needed, and also mock it in tests.

Reply:
example: https://github.com/fabric8-services/fabric8-auth/blob/master/token/keycloak/external_token.go#L26

Reply:
this I am still figuring out